lib/segment/src/index/field_index/full_text_index/compressed_posting/compressed_posting_visitor.rs

```diff
@@ -0,0 +1,175 @@
+            // check if the checked values are in the increasing order
+            if let Some(last_checked) = self.last_checked {
+                assert!(*val > last_checked);
+            }
+            self.last_checked = Some(*val);
+        }
+
+        if !self.chunk_reader.is_in_postings_range(*val) {
+            return false;
+        }
+
+        // check if current decompressed chunks range contains the value
+        if self.decompressed_chunk_idx.is_some() {
+            // check if value is in decompressed chunk range
+            // check for max value in the chunk only because we already checked for min value while decompression
+            let last_decompressed = &self.decompressed_chunk[BitPackerImpl::BLOCK_LEN - 1];
+            match val.cmp(last_decompressed) {
+                std::cmp::Ordering::Less => {
+                    // value is less than the last decompressed value
+                    return self.find_in_decompressed_and_advance(val);
+                }
+                std::cmp::Ordering::Equal => {
+                    // value is equal to the last decompressed value
+                    return true;
+                }
+                std::cmp::Ordering::Greater => {}
+            }
+        }
+
+        // decompressed chunk is not in the range, so we need to decompress another chunk
+        // first, check if there is a chunk that contains the value
+        let Some(chunk_index) = self
+            .chunk_reader
+            .find_chunk(val, self.decompressed_chunk_idx)
+        else {
+            // value is in the noncompressed postings range
+            self.decompressed_chunk_idx = None;
+            return self
+                .chunk_reader
+                .reminder_postings
+                .binary_search(val)
+                .is_ok();
+        };
+        // if the value is the initial value of the chunk, we don't need to decompress the chunk
+        if self.chunk_reader.chunks[chunk_index].initial == *val {
+            return true;
+        }
+
+        self.decompress_chunk(chunk_index);
+
+        // check if the value is in the decompressed chunk
+        self.find_in_decompressed_and_advance(val)
+    }
+
+    fn find_in_decompressed_and_advance(&mut self, val: &PointOffsetType) -> bool {
+        match self.decompressed_chunk[self.decompressed_chunk_start_index..].binary_search(val) {
+            Ok(idx) => {
+                self.decompressed_chunk_start_index = idx;
+                true
+            }
+            Err(idx) => {
+                self.decompressed_chunk_start_index = idx;
+                false
+            }
+        }
+    }
+
+    pub fn get_by_offset(&mut self, offset: usize) -> Option<PointOffsetType> {
+        let chunk_idx = offset / BitPackerImpl::BLOCK_LEN;
+
+        if chunk_idx > self.chunk_reader.chunks.len() {
+            // Offset out of range
+            return None;
+        }
+
+        if chunk_idx == self.chunk_reader.chunks.len() {
```

Comment:
The previous comment about wrong check is incomplete without this suggestion
```suggestion
        if chunk_idx >= self.chunk_reader.chunks.len() {
```